src/plugins/buflist/buflist-bar-item.c

```diff
@@ -596,6 +597,34 @@ buflist_bar_item_buflist_cb (const void *pointer, void *data,
                                    "format_lag", "");
         }
 
+        /* tls protocol */
+        hdata_irc_server = weechat_hdata_get ("irc_server");
```

Comment:
Wherever possible in the buflist, I try to be agnostic of the protocol used.
As other protocols are supported via scripts (for example Slack), we could try to be compatible with them by not checking IRC data at all and just relying on the local buffer variable you set (in IRC plugin).
The variable should already be set only if you are connected, so you could directly get the variable value for the format here, no?

Reply:
Done--I added a cleartext annotation to dcc-chat as well as a proof of concept.